runtime/autoload/health/nvim.vim

```diff
@@ -0,0 +1,410 @@
+        call health#report_warn('Your virtualenv is not set up optimally.', [
+              \ printf('Suggestion: Create a virtualenv specifically '
+              \ . 'for Neovim and use "g:%s".  This will avoid '
+              \ . 'the need to install Neovim''s Python client in each '
+              \ . 'virtualenv.', host_prog_var)
+              \ ])
+      endif
+    endif
+  endif
+
+  if empty(python_bin) && !empty(python_bin_name)
+    " An error message should have already printed.
+    call health#report_error(printf('"%s" was not found.', python_bin_name))
+  elseif !empty(python_bin) && !s:check_bin(python_bin)
+    let python_bin = ''
+  endif
+
+  " Check if $VIRTUAL_ENV is active
+  let virtualenv_inactive = 0
+
+  if exists('$VIRTUAL_ENV')
+    if !empty(pyenv)
+      let pyenv_prefix = resolve(s:trim(system(['"' . pyenv . '"', 'prefix'])))
+      if $VIRTUAL_ENV != pyenv_prefix
+        let virtualenv_inactive = 1
+      endif
+    elseif !empty(python_bin_name) && exepath(python_bin_name) !~# '^'.$VIRTUAL_ENV.'/'
+      let virtualenv_inactive = 1
+    endif
+  endif
+
+  if virtualenv_inactive
+    let suggestions = [
+          \ 'If you are using Zsh, see: http://vi.stackexchange.com/a/7654/5229',
+          \ ]
+    call health#report_warn(
+          \ '$VIRTUAL_ENV exists but appears to be inactive. '
+          \ . 'This could lead to unexpected results.',
+          \ suggestions)
+  endif
+
+  " Diagnostic output
+  call health#report_info('Executable:' . (empty(python_bin) ? 'Not found' : python_bin))
+  if len(python_multiple)
+    for path_bin in python_multiple
+      call health#report_info('Other python executable: ' . path_bin)
+    endfor
+  endif
+
+  if !empty(python_bin)
+    let [pyversion, current, latest, status] = s:version_info(python_bin)
+    if a:version != str2nr(pyversion)
+      call health#report_warn('Got an unexpected version of Python.' .
+                  \ ' This could lead to confusing error messages.')
+    endif
+    if a:version == 3 && str2float(pyversion) < 3.3
+      call health#report_warn('Python 3.3+ is recommended.')
+    endif
+
+    call health#report_info('Python Version: ' . pyversion)
+    call health#report_info(printf('%s-neovim Version: %s', python_bin_name, current))
+
+    if current == 'not found'
+      let suggestions = [
+            \ 'Use the command `pip' . a:version . 'install neovim`',
+            \ ]
+      call health#report_error(
+            \ 'Neovim Python client is not installed.',
+            \ suggestions)
+    endif
+
+    if latest == 'unknown'
+      call health#report_warn('Unable to fetch latest Neovim Python client version.')
+    endif
+
+    if status == 'outdated'
+      call health#report_warn('Latest Neovim Python client versions: ('.latest.')')
+    else
+      call health#report_ok('Latest Neovim Python client is installed: ('.status.')')
+    endif
+  endif
+
+  call health#report_notes(notes)
```

Comment:
Does the checker itself need to call this function? It should be done implicitly by `CheckHealth foo`, not the checker itself.

Then `health#report_notes` can be made private (`s:echo_report()` or whatever).